webapp/first_snap/views.py

```diff
@@ -47,11 +85,45 @@ def get_language(language):
     )
 
 
+@first_snap.route("/<language>/snapcraft.yaml")
+def get_language_snapcraft_yaml(language):
+    filename = f"first_snap/content/{language}/package.yaml"
+    snapcraft_yaml_filename = f"first_snap/content/{language}/snapcraft.yaml"
+    snap_name_cookie = f"fsf_snap_name_{language}"
+    steps = get_file(filename)
+
+    if not steps:
+        return flask.abort(404)
+
+    snap_name = steps["name"]
+
+    if snap_name_cookie in flask.request.cookies:
+        snap_name = flask.request.cookies.get(snap_name_cookie)
+
+    snapcraft_yaml = get_file(snapcraft_yaml_filename, {"${name}": snap_name})
```

Comment:
@jpmartinspt On one hand it's nice we could reuse `get_file` here and let it do the replacement. On the other hand we parse the YAML just to dump it back into text form again and download as a file.

I suppose we could do file reading and replacing without involving YAML library, but that would mean reimplementing `get_file` (just skipping YAML parsing).

Thoughts?